pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,192 @@
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useContext, useEffect, useState } from 'react';
+import cockpit from 'cockpit';
+import { Button } from '@patternfly/react-core/dist/esm/components/Button/index.js';
+import { ClipboardCopy } from '@patternfly/react-core/dist/esm/components/ClipboardCopy/index.js';
+import { EmptyState, EmptyStateBody } from '@patternfly/react-core/dist/esm/components/EmptyState/index.js';
+import { FormGroup, FormFieldGroup, FormFieldGroupHeader } from '@patternfly/react-core/dist/esm/components/Form/index.js';
+import { Grid } from '@patternfly/react-core/dist/esm/layouts/Grid/index.js';
+import { InputGroup } from '@patternfly/react-core/dist/esm/components/InputGroup/index.js';
+import { MinusIcon } from '@patternfly/react-icons';
+import { TextInput } from '@patternfly/react-core/dist/esm/components/TextInput/index.js';
+
+import { Name, NetworkModal, dialogSave } from "./dialogs-common";
+import { ModelContext } from './model-context';
+import { useDialogs } from 'dialogs.jsx';
+
+import './wireguard.scss';
+
+const _ = cockpit.gettext;
+
+export function WireGuardDialog({ settings }) {
+    const Dialogs = useDialogs();
+    const idPrefix = 'network-wireguard-settings';
+    const model = useContext(ModelContext);
+
+    const [iface, setIface] = useState(settings.connection.interface_name);
+    const [privateKey, setPrivateKey] = useState('');
+    const [publicKey, setPublicKey] = useState('');
+    const [listenPort, setListenPort] = useState('');
+    const [addresses, setAddresses] = useState('');
+    const [dialogError, setDialogError] = useState('');
+    const [peers, setPeers] = useState([]);
+
+    useEffect(() => {
+        async function getPublicKey() {
+            try {
+                const key = await cockpit.script(`echo ${privateKey.trim()} | wg pubkey`, { err: 'message' });
+                setPublicKey(key.trim());
+            } catch (e) {
+                setPublicKey('');
+            }
+        }
+
+        getPublicKey();
+    }, [privateKey]);
+
+    function onSubmit() {
+        function createSettingsObj() {
+            return {
+                connection: {
+                    id: `con-${iface}`,
+                    interface_name: iface,
+                    type: 'wireguard'
+                },
+                wireguard: {
+                    private_key: privateKey.trim(),
+                    listen_port: +listenPort,
+                    peers,
+                },
+                ipv4: {
+                    addresses: [[addresses.split('/')[0], addresses.split('/')[1], '0.0.0.0']], // TODO: fix gateway
```

Comment:
DRY, please split the address once. This should also get some validation that there the address format is valid -- there may be no /, or more than one, or an empty first/second part, etc.